spec/controllers/reservations_controller_spec.rb

```diff
@@ -168,6 +169,85 @@
       end
     end
 
+    context 'when accessed by a patron' do
+      subject { get :index }
+      it { is_expected.to be_success }
+      it { is_expected.to render_template(:index) }
+
+      before(:each) do
+        sign_in @user
+        @filter = :reserved
+
+        # clear the database, set viewing session limits
+        Reservation.delete_all
+        session[:index_start_date] = Time.zone.today
+        session[:index_end_date] = Time.zone.today + 10.days
+      end
+
+      it 'populates @reservations_set with proper date overlap' do
+        # Setup
+
+        # Enter five test reservations
+        res1 = FactoryGirl.build(
```

Comment:
You can probably do this with an array of reservations - e.g.:

```
test_reservations = []
5.times do
  test_reservations << FactoryGirl.build(...
end
```

You can then use arrays of `start_date_offsets` and `due_date_offsets` and iterate over all of the arrays to update the reservations and save them. There's probably an even nicer way to do this with hashes, but I'd have to think more about it. This would heavily clean up the tests and reduce their length, ideally without sacrificing legibility.